docs/content/about/development/code-examples.mdx

```diff
@@ -0,0 +1,176 @@
+---
+title: Using Code Examples - NCI Design System
+---
+
+# Documenting NCIDS Component Code Examples
+
+NCIDS component pages provide both guidance on when to use a component, but also **how** to use it. A key part of explaining the how is demonstrating how the HTML, or React, will appears and what markup is used to display it. We have created a `Code` component in the documentation site that allows developers to do just that.
+
+## Example of Code Blocks with previews
+
+We support previews for `html` and `jsx` examples. For JSX, this is specifically supporting `@nciocpl/ncids-react` and React core components.
+
+### Html Example
+
+---
+
+#### Markup to Use
+
+````markdown
+```html
+<div>Hello World</div>
+```
+````
+
+#### Working Example
+
+```html
+<div>Hello World</div>
+```
+
+### Html Example with NCIDS-JS initialization
+
+---
+
+HTML examples are converted to React components first then added into the DOM. During development this is done as part of the Gatsby app. For production builds, the HTML will be part of the generated page. The `<NciDsJsInit>` element can be added to your `mdx` pages to add a initialization. The expected contents of the element is the javascript you will use to initialize your JS. Due to the way we needed to implement this, there are some constraints:
+
+1. The code will be executed for **EACH** html `Code` block on the page.
+2. The element that wraps the preview will be available to your script as the variable `preview`.
+3. The code is run as a listener on the window, and most likely run before DOM content loaded. (So please do not assume anything)
+
+#### RECOMMENDATION USE
+
+- Use only one `NciDsJsInit` component per page
+- Group all preview block initialization scripts within the single `NciDsJsInit`
+- Only initialize the examples contained within the `preview` element
+  - For example use `preview.querySelectorAll('.usa-component')` to get all of your components
+
+#### Markup to Use
+
+````markdown
+```html
+<div>Open the Dev Tools Console</div>
+```
+
+<NciDsJsInit>
+	{/* eslint-disable */}
+	console.log('Html Exmaple with NCIDS-JS initialization');
+	{/* eslint-enable */}
+</NciDsJsInit>
+````
+
+> **_NOTE:_** The `{/* eslint-disable */}` `{/* eslint-enable */}` is required so that the linter does not flag all of your script as bad React.
+
+#### Working Example
+
+```html
+<div>Open the Dev Tools Console</div>
+```
+
```

Comment:
<img width="1117" alt="Screen Shot 2022-02-23 at 11 08 40 AM" src="https://user-images.githubusercontent.com/45469809/155359176-d50e6fe4-d184-4443-ad00-16a8bab9fd67.png">

Reply:
This is not an issue in the web site.